prow/tide/tide.go

```diff
@@ -1151,7 +1152,16 @@ func (c *Controller) presubmitsByPull(sp *subpool) (map[int][]config.Presubmit,
 		}
 	}
 
-	for _, ps := range c.config().Presubmits[sp.org+"/"+sp.repo] {
+	var headRefs []string
+	for _, pr := range sp.prs {
+		headRefs = append(headRefs, string(pr.HeadRefOID))
+	}
+	configPresubmits, err := c.config().GetPresubmits(c.gc, sp.org+"/"+sp.repo, sp.sha, headRefs...)
```

Comment:
This does not produce the correct set of presubmits. This assumes that all PRs in a subpool are part of a single batch and won't be merged individually or in a smaller batch.

The presubmits that are required for a set of PRs now need to be determined for every set of PRs we consider, a simple union is no longer sufficient.

Reply:
Yeah you are right, I will update this